cli/src/commands/chain_spec/mod.rs

```diff
@@ -0,0 +1,131 @@
+// Copyright 2019-2023 Parity Technologies (UK) Ltd.
+// This file is dual-licensed as Apache-2.0 or GPL-3.0.
+// see LICENSE for license details.
+
+use clap::Parser as ClapParser;
+#[cfg(feature = "chain-spec")]
+use serde_json::Value;
+use std::{io::Write, path::PathBuf};
+use subxt_codegen::fetch_metadata::Url;
+
+mod fetch;
+
+/// Download chainSpec from a substrate node.
+#[derive(Debug, ClapParser)]
+pub struct Opts {
+    /// The url of the substrate node to query for metadata for codegen.
+    #[clap(long)]
+    url: Url,
+    /// Write the output of the command to the provided file path.
+    #[clap(long, short, value_parser)]
+    output_file: Option<PathBuf>,
+    /// Replaced the genesis raw entry with a stateRootHash to optimize
+    /// the spec size and avoid the need to calculate the genesis storage.
+    ///
+    /// This option is enabled with the `chain-spec-root-hash` feature.
```

Comment:
Nit: Here I think the feature mentioned in the docs should be "chain-spec" instead of "chain-spec-root-hash".